cbsurge/stats/vector_zonal_stats.py

```diff
@@ -31,11 +31,10 @@ def vector_line_zonal_stats(df_polygon,
     df_output = df_polygon.copy()
 
     selected_operator = VECTOR_LINE_OPERATORS.get(operator)
-    assert selected_operator is not None, f"Operator '{operator}' is not supported."
 
     df_line_cloned = df_line.copy()
     intersects = gpd.sjoin(df_line_cloned, df_output, predicate="intersects", how="inner")
+    assert selected_operator is not None, f"Operator '{operator}' is not supported."
```

Comment:
This is my mistake of function implementation. could you move the below two lines before line number 31? I think it is better to exit function before doing any gpd operations in memory

```python
selected_operator = VECTOR_LINE_OPERATORS.get(operator)
assert selected_operator is not None, f"Operator '{operator}' is not supported."
```